cmd/clusterctl/client/tree/util.go

```diff
@@ -90,8 +91,13 @@ func objToSetter(obj client.Object) conditions.Setter {
 }
 
 // VirtualObject return a new virtual object.
-func VirtualObject(namespace, kind, name string) *unstructured.Unstructured {
+func VirtualObject(namespace, kind, name string, uid *string) *unstructured.Unstructured {
```

Comment:
I see why you had to do this but this would be a breaking API change. In such cases we should be creating a new function to replace with the new signature we need and add a deprecation notice on top of the `VirtualObject` function. 
Adding a comment like `// Deprecated: Use <new-function-name>. This function will be dropped in a future release.` on top of the `VirtualObject` function should be enough.

Reply:
Actually on the other hand, wdyt about adding an internal wrapper function like `createVirtualObjectWithUID()` instead so we don't have to change the function?